Casks/s/shotcut.rb

```diff
@@ -1,18 +1,32 @@
 cask "shotcut" do
-  version "24.09.13"
-  sha256 "60ac07ccf48617ca3c344175a651698f80cd73068aa3b5a3c1cfb8c2831d2177"
+  version "24.09.13,240919"
+  sha256 "fa05e60dfe7be6c78155c3ce9aae18ceb7882c40baa66770badf2dbb7c54a4ce"
 
-  url "https://github.com/mltframework/shotcut/releases/download/v#{version}/shotcut-macos-#{version.no_dots}.dmg",
+  url "https://github.com/mltframework/shotcut/releases/download/v#{version.csv.first}/shotcut-macos-#{version.csv.second || version.csv.first.no_dots}.dmg",
       verified: "github.com/mltframework/shotcut/"
   name "Shotcut"
   desc "Video editor"
   homepage "https://www.shotcut.org/"
 
+  # Upstream updates the binary versions without tagging a new release,
+  # resulting in circumstances where the tag and filename version do not match
   livecheck do
     url :url
-    strategy :github_latest
+    regex(%r{/v?(\d+(?:\.\d+)+)/shotcut[._-]macos[._-]v?(\d+(?:\.\d+)*)\.dmg$}i)
+    strategy :github_latest do |json, regex|
+      json["assets"]&.map do |asset|
+        match = asset["browser_download_url"]&.match(regex)
+        next if match.blank?
+
+        next match[1] if match[1]&.tr(".", "") == match[2]
```

Comment:
```suggestion
        next match[1] if match[1].tr(".", "") == match[2].tr(".", "")
```

The first capture group isn't optional in the regex, so it will be present if the regex matches (i.e., no need to use the safe navigation operator).

The regex will (correctly) match a filename version with dots, so we should also call `#tr` on `match[2]` to ensure that the comparison will always work as expected. If the filename version format is updated to include dots in the future, we would end up in a situation where the tag version and filename version wouldn't be equal (e.g., `123` != `1.2.3`).